src/System.Device.Gpio/System/Device/Gpio/Drivers/RaspberryPi3Driver.Linux.cs

```diff
@@ -90,7 +72,7 @@ protected internal override void AddCallbackForPinValueChangedEvent(int pinNumbe
             InitializeSysFS();
 
             _sysFSDriver.OpenPin(pinNumber);
-            _sysFSDriver.SetPinMode(pinNumber, GetModeForUnixDriver(_sysFSModes[pinNumber]));
+            _pinModes[pinNumber] = (_pinModes[pinNumber].CurrentPinMode, true);
```

Comment:
Why are we not setting the pinmode any longer? I believe that is still required right?

Reply:
I'll check again, but IIRC this would be conflicting, because we set the Pin mode using the Raspi code (via /dev/gpiomem). Like this, we can also set the Pin Mode to PullUp or PullDown, which is not supported by SysFs.

Reply:
Correct, yes. The pin must have been open at this time (or the controller would have thrown an exception). And when the pin was opened, the PinMode was already set, to any mode possible by _this_ driver, which includes pullup stuff the interrupt driver does not support (at least if it's sysfs)